crates/sui/src/sui_commands.rs

```diff
@@ -429,6 +507,218 @@ impl SuiCommand {
     }
 }
 
+/// Starts a local network with the given configuration.
+async fn start(
+    config: Option<PathBuf>,
+    with_faucet: bool,
+    with_graphql: bool,
+    with_indexer: bool,
+    random_genesis: bool,
+    epoch_duration_ms: Option<u64>,
+    indexer_rpc_port: u16,
+    fullnode_rpc_port: u16,
+    no_full_node: bool,
+    graphql_host: String,
+    graphql_port: u16,
+    faucet_port: u16,
+    pg_host: String,
+    pg_port: u16,
+    pg_db_name: String,
+    pg_user: String,
+    pg_password: String,
+) -> Result<(), anyhow::Error> {
+    if with_graphql {
+        ensure!(
+            with_indexer,
+            "GraphQL requires the indexer to be started. Please pass \
+            `--with-indexer` flag and ensure you have a Postgres DB running locally or pass in the \
+            required `pg-host` and related `pg-` flags to connect to a remote DB. For more \
+            information, see https://github.com/MystenLabs/sui/tree/main/crates/sui-graphql-rpc."
+        );
+    }
+
+    if with_indexer {
+        ensure!(
+            !no_full_node,
+            "Cannot start the indexer without a fullnode."
+        );
+    }
+
+    if epoch_duration_ms.is_some() && genesis_blob_exists(config.clone()) && !random_genesis {
+        bail!(
+            "Epoch duration can only be set when passing the `--random-genesis` flag, or when \
+            there is no genesis configuration in the default Sui configuration folder or the given \
+            config_dir.",
+        );
+    }
+
+    if with_indexer {
+        tracing::info!("Starting the indexer service at 0.0.0.0:{indexer_rpc_port}");
+    }
+    if with_graphql {
+        tracing::info!("Starting the GraphQL service at {graphql_host}:{graphql_port}");
+    }
+    if with_faucet {
+        tracing::info!("Starting the faucet service at 127.0.0.1:{faucet_port}");
+    }
+
+    let data_ingestion_path = tempdir()?.into_path();
+    let mut swarm_builder = Swarm::builder();
+
+    // If this is set, then no data will be persisted between runs, and a new genesis will be
+    // generated each run.
+    if random_genesis {
+        swarm_builder = swarm_builder.committee_size(NonZeroUsize::new(NUM_VALIDATORS).unwrap());
+        let genesis_config = GenesisConfig::custom_genesis(1, 100);
+        swarm_builder = swarm_builder.with_genesis_config(genesis_config);
+        if let Some(config) = config.clone() {
+            swarm_builder = swarm_builder.dir(config);
+        }
```

Comment:
It seems that some of the logic in this case could be merged with the case when random_genesis is false? For instance, could we just call `genesis()` here?

Reply:
If you're referring to `genesis` function in `sui_commands.rs`, that one actually creates a genesis and saves it locally on disk. I will rework this, I think it's not quite right but I need to try a few things out.

Reply:
I think this was wrong, so I now fixed it. If a config dir is passed, then we go with what's there. If both force-regenesis and config-dir are passed, then it will complain and error.